From: Fab Stz <fabstz-it@yahoo.fr>
Date: Thu, 6 Jul 2017 11:29:02 +0200
Subject: trailing colon question mark

Sometimes, the files of the ISO when mounted with fuseiso have a trailing
";?". Depending on the way fuseiso mounted the ISO (-c <charset>) option the
question mark '?' may be different, especially when mounted on with utf-8.

Better detected and manage this case.

See https://bugs.debian.org/867388
---
 vobcopy.c | 25 +++++++++++++++++++------
 1 file changed, 19 insertions(+), 6 deletions(-)

diff --git a/vobcopy.c b/vobcopy.c
index d7260d0..d7e53cc 100644
--- a/vobcopy.c
+++ b/vobcopy.c
@@ -1026,10 +1026,16 @@ next: /*for the goto - ugly, I know... */
                 }
               else
                 {
-                  if( strstr( d_name, ";?" ) )
-                    {
-                      fprintf( stderr, _("\n[Hint] File on dvd ends in \";?\" (%s)\n"), d_name );
-                      strncat( output_file, d_name, strlen( d_name ) - 2 );
+                  if( strstr( d_name, ";" ) )
+                  {
+                      char * pch;
+                      int position_from_end;
+                      pch = strrchr(d_name, ';');
+                      position_from_end = strlen( d_name ) - (pch - d_name);
+                      if ( position_from_end < 4 ) {
+                        fprintf( stderr, _("\n[Hint] File on dvd ends in \";?\" (%s)\n"), d_name );
+                        strncat( output_file, d_name, strlen( d_name ) - position_from_end );
+                      }
                     }
                   else
                     {
@@ -1274,8 +1280,15 @@ next: /*for the goto - ugly, I know... */
 
                       for( a = 1; a < subvob; a++ )
                         {
-                          if( strstr( input_file, ";?" ) )
-                            input_file[ strlen( input_file ) - 7 ] = ( a + 48 );
+                          if( strstr( input_file, ";" ) )
+                            {
+                              char * pch;
+                              int position_from_end;
+                              pch = strrchr( input_file, ';' );
+                              position_from_end = strlen( input_file ) - ( pch - input_file );
+                              if ( position_from_end < 4 )
+                                input_file[ strlen( input_file ) - 5 - position_from_end ] = ( a + 48 );
+                            } 
                           else
                             input_file[ strlen( input_file ) - 5 ] = ( a + 48 );
 
